drivers/i2c/i2c_ll_stm32_v2.c

```diff
@@ -600,28 +716,37 @@ static int stm32_i2c_msg_write(const struct device *dev, struct i2c_msg *msg,
 	data->current.is_err = 0U;
 	data->current.msg = msg;
 
+#ifdef CONFIG_DCACHE
+	if (!buf_in_nocache((uintptr_t)msg->buf, msg->len)) {
+#ifdef STRICT_MEMORY_CHECK
```

Comment:
I'm not aware about this option. To be cleaned up unless I've missed something.

Reply:
added for my own use case, removed here